google/resource_compute_attached_disk.go

```diff
@@ -0,0 +1,227 @@
+			Create: schema.DefaultTimeout(300 * time.Second),
+			Delete: schema.DefaultTimeout(300 * time.Second),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"attached_disk": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"attached_instance": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAttachedDiskCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	attachedInstance := d.Get("attached_instance").(string)
+	zone, err := getZoneForAttachedDisk(d, config, attachedInstance)
+	if err != nil {
+		return err
+	}
+
+	instanceName := GetResourceNameFromSelfLink(attachedInstance)
+	diskName := GetResourceNameFromSelfLink(d.Get("attached_disk").(string))
+
+	attachedDisk := compute.AttachedDisk{
+		DeviceName: diskName,
+		Source:     fmt.Sprintf("/projects/%s/zones/%s/disks/%s", project, zone, diskName),
+	}
+
+	op, err := config.clientCompute.Instances.AttachDisk(project, zone, instanceName, &attachedDisk).Do()
+	if err != nil {
+		return err
+	}
+
+	// TODO (chrisst) change format of the internal id to include project/zone
+	d.SetId(fmt.Sprintf("%s:%s", instanceName, diskName))
+
+	waitErr := computeSharedOperationWaitTime(config.clientCompute, op, project,
+		int(d.Timeout(schema.TimeoutCreate).Minutes()), "disk to attach")
+	if waitErr != nil {
+		d.SetId("")
+		return waitErr
+	}
+
+	return resourceAttachedDiskRead(d, meta)
+}
+
+func resourceAttachedDiskRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+	d.Set("project", project)
+
+	attachedInstance := d.Get("attached_instance").(string)
+	zone, err := getZoneForAttachedDisk(d, config, attachedInstance)
+	if err != nil {
+		return err
+	}
+	d.Set("zone", zone)
+
+	instanceName := GetResourceNameFromSelfLink(attachedInstance)
+	diskName := GetResourceNameFromSelfLink(d.Get("attached_disk").(string))
+
+	instance, err := config.clientCompute.Instances.Get(project, zone, instanceName).Do()
+	if err != nil {
+		return err
+	}
+
+	// Iterate through the instance's attached disk as this is the only way to
+	// confirm the disk is actually attached
+	ad := findDiskByName(instance.Disks, diskName)
+	if ad == nil {
+		// Disk was not found attached to the referenced instance
+		d.SetId("")
+		return nil
+	}
+
+	// Force the referenced resources to a self-link in state because it's more specific then name.
+	d.Set("attached_instance", instance.SelfLink)
```

Comment:
The [relative link](https://github.com/terraform-providers/terraform-provider-google/blob/10ec7b2ca9a5d191f2eb14b577f2202e2349f6f8/google/self_link_helpers.go#L56) is, I believe, better, as it won't have any diff complications across versions of the API. Though honestly my preference here is weak.

Reply:
Or self link sets can be wrapped with `ConvertSelfLinkToV1`

Reply:
Looks easy enough, I'm down